Adding a blog post about React Router brand updates (#304)
remix-run/remix-website: data/posts/react-router-brand-update.md

---
title: React Router Brand Update
summary: "The evolution of the React Router brand identity in conjunction with the release of v7."
featured: false
date: 2024-10-29
image: /blog-images/headers/react-router-brand-update.jpg
ogImage: /blog-images/headers/react-router-brand-update.jpg
imageAlt: React Router Brand Update
imageDisableOverlay: true
authors:
  - Tim Quirino
---

> Since I graduated from college before IPhones existed, I was steeped in ancient texts predating everything we know about the modern web. Translation: *I was trained as a print designer.* Outside of dating myself, what I'm actually trying to say is that the future always benefits from a rich history of experiences and references. If you pay close attention to fashion and trends, you know this to be true. Old things frequently inspire new things, and while the Remix team has been hard at work to release React Router v7, it has been my task to update the identity of this particular new thing.

Reviewer: ```suggestion
My formal design training predates much of what we now know about the modern web. In school, I was steeped in ancient texts and trained as a print designer where I learned that the future always benefits from a rich history of experiences and references. If you pay close attention to fashion and trends, you know this to be true. Old things frequently inspire the new, and while the Remix team has been hard at work shipping React Router v7, we decided to take a look at the brand identity to see if we could freshen it up a bit.
```

<img alt="A visual summary of updates to the React Router brand identity, with visible guidelines" src="/blog-images/posts/react-router-brand-update/update-guidelines.png" class="my-8 border rounded-md" />

## Updating the logo

<img alt="Side by side comparison of the React Router logo before and after the update" src="/blog-images/posts/react-router-brand-update/logo-before-and-after.png" class="my-8 border rounded-md" />

The React Router logo is already an elegant abstraction of what React Router does. For anyone that doesn't know, it's a visual representation of a route tree. The connected nodes are an actively rendered route. There is meaning in the way it flows from top-to-bottom and connects adjacent nodes to present the quickest and cleanest path. If you've read [Matt Brophy's post about Fog of War](https://remix.run/blog/fog-of-war), you can see the obvious similarities.

> I admire it's simplicity, but from an execution perspective, there were obvious opportunities for improvement. 

Reviewer 2: ```suggestion
I admire its simplicity, but from an execution perspective, there were obvious opportunities for improvement. 
```

<img alt="The old React Router logo with guidelines showing what to fix" src="/blog-images/posts/react-router-brand-update/logo-before.png" class="my-8 border rounded-md" />

> Nodes were misaligned, inconsistently positioned, and upon closer inspection it was easy to see that the curves in their connective tissue were not smooth. While I spent some time exploring alternative logos, I came to realize that fixing these issues were the top priority.  

Reviewer 2: ```suggestion
Nodes were misaligned, inconsistently positioned, and upon closer inspection it was easy to see that the curves in their connective tissue were not smooth. While I spent some time exploring alternative logos, I came to realize that fixing these issues were the top priority.  
```

Ultimately, we decided against visually integrating it. The Remix brand will stay separate, free to evolve as we awaken from our nap in 2025 to release new Remix packages. By making this decision, the React Router wordmark wouldn't depend on another logo (and wouldn't limit this exploration either). It allowed us to explore something more specific to React Router. 

<img alt="Three iterations of the React Router wordmark" src="/blog-images/posts/react-router-brand-update/wordmark-iterations.png" class="m-auto w-4/5 border rounded-md" />

Another problem was simply that the words "React Router" were quite long when included in a brand lockup. Instead of creating various vertically and horizontally stacked configurations of the Logo + Wordmark, we wondered if we could provide a single, evergreen solution.

> Inspired by the ancient "compact disc" logo, we embraced a method of nesting the word "React" and emphasizing "Router" as the more significant idea. Visually, this was an elegant way of reconciling the long width issue while establishing a more accurate hierarchy for the identity.

Reviewer 2: Worth including a link here for the youngins?

Reviewer 2: Or even an image of the Compact Disk logo underneath the line

> When the word "Router" is styled as an outline, you can more clearly see the retro inspiration behind it.

Might be distracting though 🤷

Author: I think it might be too distracting to actually include an image of the Compact Disk logo, but I'm down with adding a link!

I kept some usage of Founders Grotesk and created a custom ligature that more comfortably allowed for nesting. Connecting the "u" with the "t" is also a visual reference to the way nodes connect in the logo.

<img alt="A closeup of the React Router wordmark showing the custom ligature" src="/blog-images/posts/react-router-brand-update/wordmark.png" class="m-auto w-4/5 border rounded-md" />

<img alt="Stylistic variants for the React Router wordmark in black and white" src="/blog-images/posts/react-router-brand-update/wordmark-variants.png" class="m-auto w-4/5 border rounded-md" />

When the word "Router" is styled as an outline, you can more clearly see the retro inspiration behind it.

## Creating a mark for this release

> Initially, I didn't think we needed a unique visual identifier for "v7." After all, React Router has been around for years and previous versions never received a visual identity of their own. But a release such as this is significant to those 7+ million React Router projects (and the 5 million line Shopify App) and we take our work very seriously. 

Reviewer 2: > 7+ million

9 million on GitHub now:

<img width="350" alt="image" src="https://github.com/user-attachments/assets/0c761b80-346c-44f4-94e1-18988633b760">

I think you could just say the millions. It's so hard to quantify this stuff, especially because this doesn't capture private repos, and it does capture numerous demo apps that don't _really_ count

Author: Ah, great to know! (9.1million) Up 29% from ~7million is non-trivial, so "millions" doesn't quite have the same impact. Then again, I'm not trying to inflate the significance of the point using the difficult-to-quantify demo apps either, so perhaps "millions" is the better edit.

I included the evolution of the "v7" ligature to illustrate where I initially saw an opportunity. When initially typing the characters "v" and "7", the default letter spacing made them feel like separate objects at an uncomfortable distance apart. 

When kerning them together, nesting started to emerge as an obvious direction. There were nested elements in the wordmark, so nesting elements here seemed conceptually sound as well. The task of refining this was then an issue of finding a balance between keeping the characters consistent with the proportions for the rest of the lockup, while customizing the characters to feel like they *belong together.*

<img alt="A closeup of the custom v7 ligature" src="/blog-images/posts/react-router-brand-update/v7-ligature.png" class="m-auto w-4/5 border rounded-md" />

<img alt="All three updates to the React Router brand identity against a white background" src="/blog-images/posts/react-router-brand-update/update.png" class="m-auto w-4/5 border rounded-md" />

> I'm happy with the way this brand update turned out because the process is a reflection of the way the Remix team works. Far from being linear, this task meandered playfully without compromising detail and intentionality. Perhaps more importantly, it's a true evolution—one that allows us to be more flexible and creative about our identity—without losing sight of how we got here.

Reviewer 2: Poetry

> <img alt="Six badges designed in the style of the old web to show how new React Router brand updates can be used" src="/blog-images/posts/badges.png" class="m-auto w-4/5 border rounded-md" />

Reviewer 2: This link is broken

```suggestion
<img alt="Six badges designed in the style of the old web to show how new React Router brand updates can be used" src="/blog-images/posts/react-router-brand-update/badges.png" class="m-auto w-4/5 border rounded-md" />
```
